src/context/schedule.tsx

```
@@ -29,18 +30,64 @@
 
+  // Optimized schedule update handler
+  const handleScheduleUpdate = useCallback((newSchedule: ScheduleItem[]) => {
+    loadSchedule(newSchedule);
+  }, [loadSchedule]);
+
+  // Memoized addToSchedule function to prevent unnecessary re-renders
+  const memoizedAddToSchedule = useCallback(async (
+    dateTime: Date,
+    url: string,
+    room?: string
+  ): Promise<void | DocumentReference<DocumentData>> => {
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)